exchanges/stream/websocket.go

```diff
@@ -854,6 +848,19 @@ func (w *Websocket) SubscribeToChannels(channels []ChannelSubscription) error {
 	return w.Subscriber(channels)
 }
 
+// fullSubscribeToChannels is required to transmit full subscription. To keep
+// subscriptions active already subscribed checks are bypassed this is an edge
+// case utilised by the gateio websocket subscription endpoint.
+func (w *Websocket) fullSubscribeToChannels(channels []ChannelSubscription) error {
+	if len(channels) == 0 {
+		return fmt.Errorf("%s websocket: cannot subscribe no channels supplied",
+			w.exchangeName)
+	}
+	w.subscriptionMutex.Lock()
+	defer w.subscriptionMutex.Unlock()
+	return w.Subscriber(channels)
+}
+
```

Comment:
I think this could be solved without this function by not returning an error if something is already subscribed.
See here: 
https://github.com/thrasher-corp/gocryptotrader/pull/541/files#diff-c248a1ef42c669be711d82b5cadb3016R842

This would also fix another bug where if subscription 1/100 in the list of channels is already subscribed, then 99 channels don't get to subscribe because you're returning on error

Reply:
I see your point, but if we do get an error that means that subscription generation/regeneration for an exchange is broken. I would rather keep the error for implementation/integration issues for future devs. Debatable issue. @xtda @thrasher-